databricks/koalas/frame.py

```diff
@@ -2355,10 +2361,33 @@ def _pd_getitem(self, key):
         raise NotImplementedError(key)
 
     def __repr__(self):
-        return repr(self.head(max_display_count).to_pandas())
+        pdf = self.head(max_display_count).to_pandas()
+        repr_string = repr(pdf)
+        if len(pdf) == max_display_count:
```

Comment:
haha the trick is to call head on max_display_count + 1 records, and compare with that.

In that case, if there's exactly max_display_count records, we won't show that message which could be misleading.